tests/e2e/minikube-kubernetes-istio/README.md

```diff
@@ -0,0 +1,65 @@
+# Benefits:
+1. Set up a local minikube VM Environment once and run E2E tests on local machine, so you can test and debug locally.
+1. No need to worry about kubernetes cluster setup.
+
+# Prereqs:
+1. Set up Istio Dev envrionment using https://github.com/istio/istio/wiki/Dev-Guide.
+
+1. Install
+  * [kvm2 for linux](https://www.linux-kvm.org/page/Main_Page) 
+  * [docker](https://docs.docker.com/) - Verify `docker version` returns version >= 18.03.0-ce
+  * [minikube](https://www.vagrantup.com/downloads.html) - Verify `minikube version` returns version >= Vagrant 0.25.0
+  * [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl) - Verify `kubectl version` returns both server and client versions
+  * [curl](https://curl.haxx.se/) - Verify `curl --help` prints the help information.
+
+You can run the following script to check/install of all pre-requisites, or use it as a reference to install them manually.
+(This requires installation of [Homebrew](https://brew.sh) on macOS or debian based Linux distributions)
+
+```bash
+sh install_prereqs_debian.sh
+```
+
+# Steps
+## 1. Set up Minikube Environment
+```bash
+sh setup_host.sh
+```
+
+## 2. Build istio images
```

Comment:
@kimikowang Perhaps you can incorporate your local registry here and simplify the push step?